src/main/scala/org/apache/spark/sql/delta/DeltaAnalysis.scala

```diff
@@ -59,14 +59,20 @@ class DeltaAnalysis(session: SparkSession, conf: SQLConf)
       }
 
     // INSERT OVERWRITE by ordinal
-    case a @ OverwriteByExpression(
-        DataSourceV2Relation(d: DeltaTableV2, _, _, _, _), _, query, _, false)
+    case o @ OverwriteByExpression(
+        DataSourceV2Relation(d: DeltaTableV2, _, _, _, _), deleteExpr, query, _, false)
       if query.resolved && needsSchemaAdjustment(d.name(), query, d.schema()) =>
       val projection = normalizeQueryColumns(query, d)
       if (projection != query) {
-        a.copy(query = projection)
+        val aliases = AttributeMap(query.output.zip(projection.output).collect {
```

Comment:
I'd call it `attrMapping`

Reply:
This is an attribute map to store an origin attribute and its alias, so we can fastly understand the line `case a: AttributeReference => aliases.getOrElse(a, a)`